pkg/compose/hash.go

```diff
@@ -39,3 +40,28 @@ func ServiceHash(o types.ServiceConfig) (string, error) {
 	}
 	return digest.SHA256.FromBytes(bytes).Encoded(), nil
 }
+
+// ServiceDependenciesHash computes the configuration hash for service dependencies.
+func ServiceDependenciesHash(project *types.Project, o types.ServiceConfig) (string, error) {
+	bytes := make([]byte, 0)
+	for _, serviceConfig := range o.Configs {
+		projectConfig, ok := project.Configs[serviceConfig.Source]
+		if !ok {
+			continue
+		}
+
+		if projectConfig.Content != "" {
+			bytes = append(bytes, []byte(projectConfig.Content)...)
+		} else if projectConfig.File != "" {
+			content, err := os.ReadFile(projectConfig.File)
```

Comment:
`File` could also be a folder. Maybe an option would be to read content into a tar (code exists to inject secrets) and compute hash from it

While the general approach looks good, I'd prefer we have a more explicit hash and label _per resource_, and a dedicated `hash` function for configs (and secrets), so that it make it easier to diagnose use of those labels